src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdStateUpdateService.java

```diff
@@ -0,0 +1,139 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus;
+import uk.gov.hmcts.reform.sscs.reference.data.mappings.CancellationReason;
+
+import javax.validation.Valid;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.State.DORMANT_APPEAL_STATE;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.State.HEARING;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.State.READY_TO_LIST;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.State.UNKNOWN;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus.AWAITING_LISTING;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus.LISTED;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingStatus.CANCELLED;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingStatus.FIXED;
+
+
+@Slf4j
+@Service
+public class CcdStateUpdateService {
+
+    public void updateListed(HearingGetResponse hearingResponse, @Valid SscsCaseData sscsCaseData) {
+        State state = UNKNOWN;
+        if (isHearingListingStatusFixed(hearingResponse)) {
+            state = mapHmcCreatedOrUpdatedToCcd(hearingResponse);
+        }
+
+        if (isKnownState(state)) {
+            sscsCaseData.setState(state);
+            log.info("CCD state has been updated to {} for caseId {}", state, sscsCaseData.getCcdCaseId());
+        }
+    }
+
+    public void updateCancelled(HearingGetResponse hearingResponse, @Valid SscsCaseData sscsCaseData)
+            throws UpdateCaseException {
+        State state = UNKNOWN;
+
+        if (isHearingCancelled(hearingResponse)) {
+            state = mapHmcCancelledToCcdState(
+                hearingResponse.getHearingResponse().getHearingCancellationReason(), sscsCaseData.getCcdCaseId());
+        }
+        if (isKnownState(state)) {
+            sscsCaseData.setState(state);
+            log.info("CCD state has been updated to {} for caseId {}", state, sscsCaseData.getCcdCaseId());
+        }
+    }
+
+    public void updateFailed(SscsCaseData sscsCaseData) {
+        sscsCaseData.setState(State.HANDLING_ERROR);
+        log.info("CCD state has been updated to {} for caseId {}", State.HANDLING_ERROR, sscsCaseData.getCcdCaseId());
+    }
+
+    private State mapHmcCreatedOrUpdatedToCcd(HearingGetResponse hearingResponse) {
+        if (checkHearingStatus(hearingResponse, LISTED)) {
+            return HEARING;
+        }
+        if (checkHearingStatus(hearingResponse, AWAITING_LISTING)) {
+            return READY_TO_LIST;
+        }
+        return UNKNOWN;
```

Comment:
This was raised before but we need to be careful about this. We should alert whenever we are in a bad state